lib/Serialization/Deserialization.cpp

```diff
@@ -4473,10 +4473,12 @@ llvm::Error DeclDeserializer::deserializeDeclCommon() {
 
         Expected<Type> deserialized = MF.getTypeChecked(typeID);
         if (!deserialized) {
-          if (deserialized.errorIsA<XRefNonLoadedModuleError>()) {
+          if (deserialized.errorIsA<XRefNonLoadedModuleError>() ||
+              MF.allowCompilerErrors()) {
```

Comment:
Yea, it's a good idea to also cover the `XRefNonLoadedModuleError` cases for `MF.allowCompilerErrors`.